### ISSUE TITLE
Game keeps crashing

### ISSUE THREAD

**Issue by TheHunterManX** [CleverRaven/Cataclysm-DDA#13863]
Every time I try to open the game it opens for half a second then crashes.
I have looked at the log:

-----------------------------------------
14:18:35.234 : Starting log.
14:18:35.703 SDL : src/sdltiles.cpp:315: Failed to select render target: That operation is not supported
14:18:35.703 : initscr failed!
14:18:35.719 : Log shutdown.
-----------------------------------------

What do I need to play this game? I am using windows xp sp3 with directx 0.9c and Intels graphics card. Is it hardware or software that I need extra of?

**Comment by DanmakuDan** [comment]
If you have a options.txt in your config folder, try setting ```SOFTWARE_RENDERING``` to ```true```.

**Comment by namarrgon** [comment]
The windows drivers of some intel GPUs don't support a high enough OpenGL version.

**Comment by drbig** [comment]
This isn't new... Isn't the:
> try setting SOFTWARE_RENDERING to true.

documented anywhere? Not having that somewhere user-visible would certainly make this a 'bug'.

**Comment by DanmakuDan** [comment]
The game should be able to switch to software rendering automatically to avoid having to look for a solution online.

**Comment by drbig** [comment]
@DanmakuDan Yes, that would be the best solution "to rule'em all" (...feel free to PR?...).

**Comment by namarrgon** [comment]
Looking at [sdltiles.cpp](https://github.com/CleverRaven/Cataclysm-DDA/blob/d56e624ed03b6efba398b17b53ed5dda2b027d56/src/sdltiles.cpp#L334-L337) there seems to be already a fallback mechanism in place, though, not very effective.

Going by https://wiki.libsdl.org/SDL_CreateRenderer it should be possible to let sdl2 figure out the best renderer by initializing it with:
```renderer = SDL_CreateRenderer( window, -1, 0);```
instead of [manually](https://github.com/CleverRaven/Cataclysm-DDA/blob/d56e624ed03b6efba398b17b53ed5dda2b027d56/src/sdltiles.cpp#L328-L345) choosing:

```renderer = SDL_CreateRenderer( window, -1, SDL_RENDERER_ACCELERATED)```
or

```renderer = SDL_CreateRenderer( window, -1, SDL_RENDERER_SOFTWARE)```

I haven't figured out a way to force the environment to emulate software-rendering so I couldn't test it yet.

**Comment by DanmakuDan** [comment]
The hardware renderer is probably allowed for DirectX 9, but it might not support the render target feature.

The error text comes from here:
```
    if( SDL_SetRenderTarget( renderer, display_buffer ) != 0 ) {
        dbg( D_ERROR ) << "Failed to select render target: " << SDL_GetError();
        return false;
    }
```

Looks like the display buffer and render target commands should be done inside the SDL_CreateRenderer logic to make sure the texture options are supported.

**Comment by kevingranade** [comment]
When I looked at this previously, I came to the conclusion that some
drivers are lying about their capabilities to SDL, at which point I stopped
looking.

The only thing I can think of is somehow detecting that something went
wrong and falling back to software rendering on next program launch.

**Comment by TheHunterManX** [comment]
Fixed and Closed! I locked in the options.txt and found "Software Renderer False" so I changed it from False to True and now it works!
Original text
SOFTWARE_RENDERING false
Changed text
SOFTWARE_RENDERING true

**Comment by remyroy** [comment]
@TheHunterManX Would you help fix this problem for the future versions and the future users in your situation? We only need someone who has this problem to test new version that includes a fix which will not require manually editing the options.txt file.

You can download this new version on http://remyroy.com/cataclysmdda-0.C-7053-g3a04c78.zip . When you start the game, it will show 0.C as the version but that is wrong. This version was compiled from https://github.com/DanmakuDan/Cataclysm-DDA/tree/sdl_adjust_init_logic . It is based on a recent version of experimental.

Obviously, there are risks involved in running random binaries from the internet. The zip file hashes should be:

* MD5: ECF7DEC0F8887A05B9DF652C8E7AD6AB
* SHA-1: 567E36440E5E3073E7C0F6C33F9E944ADB85F223

**Comment by TheHunterManX** [comment]
The One I downloaded... Worked! No options.txt editing required!
My theory why it didnt work: (My Opengl was too old for the game to run on without software rendering(Directx) on)
Full Computer Status:
Windows Xp Professional
Version 2002
Service Pack 3

Graphics card:
Intel Graphics accelerator 3150
Opengl 1.4
Pixel Shaders 2.0

Software for graphics:
DirectX Run End Time (9.0c)

**Comment by remyroy** [comment]
This can be closed. It was solved in #13881.
